nucleus-api/src/main/java/io/github/nucleuspowered/nucleus/api/query/QueryDataCache.java

```diff
@@ -0,0 +1,32 @@
+package io.github.nucleuspowered.nucleus.api.query;
+
+/**
+ * A data cache which stores objects constructed from data retrieved from a query.
+ *
+ * @param <K> The type of the key used to retrieve cached objects.
+ * @param <V> The type of the object being cached.
+ */
+public interface QueryDataCache<K, V> {
```

Comment:
By data manager I mean the main class that handles retrieving and building data queried from the database. I believe I called the class `TicketDataManager` if you want to take a look.

Here's a diagram I drew up to help explain my use case http://m.imgur.com/AlDf7Ig.